pallets/gear/src/lib.rs

```diff
@@ -1366,13 +1367,24 @@ pub mod pallet {
             dest: &T::AccountId,
             amount: Self::Balance,
         ) -> Result<(), DispatchError> {
-            let _ = <T as Config>::Currency::repatriate_reserved(
+            let leftover = <T as Config>::Currency::repatriate_reserved(
                 &<T::AccountId as Origin>::from_origin(source),
                 dest,
                 amount,
                 BalanceStatus::Free,
             )?;
 
+            if leftover > 0_u128.unique_saturated_into() {
+                log::debug!(
+                    target: "essential",
+                    "Reserved funds not fully repatriated from 0x{}… to 0x{}… : amount = {:?}, leftover = {:?}",
+                    HexDisplay::from(&source.as_ref()[..6].to_vec()),
```

Comment:
Why don't use common rust mech instead of HexDisplay?
We have support for precisions in our display and debug implementations